### PR TITLE
Implement skinning for slider border thickness

### LINKED ISSUE #4422: Allow for the thickness of slider borders to be changed
This is basically just a feature from McOsu that I'd like to see in the full game. In McOsu you can change the thickness of the slider borders. I think it would be great if we could have that in the full game, either as an option in the menu or as a skin.ini value. Perhaps slider borders could even be fully skinned like followpoints, but maybe that's too much.
I've attached an image comparing slider border sizes, on the left is a regular thickness slider, on the right is a slider at 50% thickness
image

### PR DESCRIPTION
**Pull request by santerinogelainen**
resolves #4422

Legacy parsing might be useless since legacy client doesn't have that skin.ini option implemented afaik, but i still added it.

Size is given as a number (1 = default, 0.2 = 20% of the default, 2 = 200% of the default, etc.).

Min is 0 (= no border). Max is 8 (the point where it's just all border)

### CODE REVIEW

**Comment by peppy** [comment]
do we use int percentages like this anywhere else? also no limiting? what happens if you set to zero or 1000000?

File: osu.Game.Rulesets.Osu/Objects/Drawables/Pieces/SliderBody.cs [hunk 1]
```diff
@@ -64,6 +64,21 @@ public new Color4 BorderColour
             }
         }
 
+        /// <summary>
+        /// Used to size the path border.
+        /// </summary>
+        public int BorderSize {
```
Comment (peppy) [inline osu.Game.Rulesets.Osu/Objects/Drawables/Pieces/SliderBody.cs]: B races go in new line

**Review by peppy — CHANGES_REQUESTED** [review]

File: osu.Game.Rulesets.Osu/Objects/Drawables/Pieces/SliderBody.cs [hunk 2]
```diff
@@ -130,12 +145,28 @@ public Color4 AccentColour
...
+                    if (borderSize == value)
+                        return;
+
+                    borderSize = value;
+
+                    InvalidateTexture();
+                }
+            }
+
+            private float calucatedBorderPortion => BorderSize / 100f * border_portion;
```
Comment (swoolcock) [inline osu.Game.Rulesets.Osu/Objects/Drawables/Pieces/SliderBody.cs]: Typo.

**Comment by peppy** [comment]
Generally I'd still prefer to add new skinning features before all existing ones have been completed (and in the process, the skin system fleshed out).

For instance, this one is under `[General]` because it has no better place to be, but such settings should be per-ruleset and adjustable in-game, not just from skin.ini (both parts of a larger refactor of skinning coming in the future).

**Review by peppy — APPROVED** [review]

**Review by peppy — APPROVED** [review]

**Pull request merged**